### PR TITLE
[ADD] account_invoice_open_sale: Take links via origin into account

### PR DESCRIPTION
**Pull request by hbrunn**
I find the calculation of the flag very expensive now, should we maybe show the button unconditionally?

### CODE REVIEW

File: account_invoice_open_sale/models/account_invoice.py [hunk 1]
```diff
@@ -23,15 +23,54 @@ def _compute_has_sale_order(self):
         JOIN sale_order_line_invoice_rel AS rel ON ail.id = rel.invoice_line_id
         WHERE ail.invoice_id IN %s
         """, (tuple(self.ids),))
-
         invoice_ids = [row[0] for row in self.env.cr.fetchall()]
+
+        # invoices might also be linked via sale order's name in origin
+        order_names = []
+        origins = list(filter(None, self.mapped('origin')))
```
Comment (StefanRijnhart) [inline account_invoice_open_sale/models/account_invoice.py]: Split origins by comma and strip I think.

File: account_invoice_open_sale/models/account_invoice.py [hunk 2]
```diff
@@ -23,15 +23,54 @@ def _compute_has_sale_order(self):
...
+            order_names = [name for name, in self.env.cr.fetchall()]
+
+        # or it might be a refund of an invoice linked to an order
+        self.env.cr.execute("""
+        SELECT ai_refund.id
+        FROM account_invoice_line AS ail
+        JOIN sale_order_line_invoice_rel AS rel ON ail.id = rel.invoice_line_id
+        JOIN account_invoice ai ON ai.id=ail.invoice_id
+        JOIN account_invoice ai_refund ON ai.number=ai_refund.origin
+        WHERE ai_refund.id IN %s AND ai_refund.journal_id in %s
```
Comment (StefanRijnhart) [inline account_invoice_open_sale/models/account_invoice.py]: `ai_refund.journal_id = ai.journal_id`? https://github.com/odoo/odoo/blob/12.0/addons/sale/models/sale.py#L82

**Comment by StefanRijnhart** [comment]
About the performance, maybe we should revert to your suggestion to keep a stored (computed?) many2many of 'extra invoices` on sale orders.

**Comment by hbrunn** [comment]
back to saved many2many, this only has impact on invoice creation, and then not a lot

**Comment by StefanRijnhart** [comment]
Bumping merge commit here: https://github.com/VanMoof/odoo-buildout/pull/244

**Pull request merged**